swim/labels_test.go

```diff
@@ -0,0 +1,298 @@
+
+	err = DefaultLabelOptions.validateLabel(fixtureTooBigLabelsWithPrivate, "newkey", "some value")
+	assert.Error(t, err, "expected an error when adding a new key to a map that is over capacity (with private)")
+
+	err = DefaultLabelOptions.validateLabel(fixtureTooBigLabels, "key1", "new value for 1")
+	assert.NoError(t, err, "expected no error when overwriting a key on a map that is over capacity")
+
+	err = DefaultLabelOptions.validateLabel(fixtureTooBigLabelsWithPrivate, "key1", "new value for 1")
+	assert.NoError(t, err, "expected no error when overwriting a key on a map that is over capacity (with private)")
+
+	// make sure that private labels are still accepted to allow for ringpop internals to proceed
+	err = DefaultLabelOptions.validateLabel(fixtureLabelsRoom, "__private2", "private value2")
+	assert.NoError(t, err, "expected no error when adding a private label when there is room in the map")
+
+	err = DefaultLabelOptions.validateLabel(fixtureLabelsRoomWithPrivate, "__private2", "private value2")
+	assert.NoError(t, err, "expected no error when adding a private label when there is room in the map (with private)")
+
+	err = DefaultLabelOptions.validateLabel(fixtureExactLabels, "__private2", "private value2")
+	assert.NoError(t, err, "expected no error when adding a private label to a map that is at capacity")
+
+	err = DefaultLabelOptions.validateLabel(fixtureExactLabelsWithPrivate, "__private2", "private value2")
+	assert.NoError(t, err, "expected no error when adding a private label to a map that is at capacity (with private)")
+
+	err = DefaultLabelOptions.validateLabel(fixtureTooBigLabels, "__private2", "private value2")
+	assert.NoError(t, err, "expected no error when adding a private label when the map is over capacity")
+
+	err = DefaultLabelOptions.validateLabel(fixtureTooBigLabelsWithPrivate, "__private2", "private value2")
+	assert.NoError(t, err, "expected no error when adding a private label when the map is over capacity (with private)")
+
+	// tests for key and value size exceeding
+	err = DefaultLabelOptions.validateLabel(fixtureLabelsRoom, fixtureExceedKeySize, "some value")
+	assert.Error(t, err, "expected error when the key length exceeds the configured length")
+
+	err = DefaultLabelOptions.validateLabel(fixtureLabelsRoom, "newkey", fixtureExceedValueSize)
+	assert.Error(t, err, "expected error when the value length exceeds the configured length")
+
+	err = DefaultLabelOptions.validateLabel(fixtureLabelsRoom, fixtureExceedKeySizePrivate, "some value")
+	assert.NoError(t, err, "expected no error when the key length exceeds the configured length for a private key")
+
+	err = DefaultLabelOptions.validateLabel(fixtureLabelsRoom, fixtureExceedKeySizePrivate, fixtureExceedValueSize)
+	assert.NoError(t, err, "expected no error when the value length exceeds the configured length for a private key")
+}
+
+func TestLabelOptions_validateLabels(t *testing.T) {
```

Comment:
non-blocking:  This are a lot of tests; what i'm missing is adding multiple non-private or private labels (you only test one label or a combination of private and non-private) :)